src/main/java/com/example/webapp/Service/DepartmentService.java

```diff
@@ -0,0 +1,64 @@
+package com.example.webapp.Service;
+
+import com.example.webapp.Exception.DepartmentNotFoundException;
+import com.example.webapp.Exception.EmployeeNotFoundException;
+import com.example.webapp.Model.Department;
+import com.example.webapp.Repository.DepartmentRepository;
+import com.example.webapp.dto.DepartmentDTO;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static com.example.webapp.Model.Department.convertDTOtoENT;
+import static com.example.webapp.Model.Department.convertENTtoDTO;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class DepartmentService {
+
+    private final DepartmentRepository departmentRepository;
+
+    public List<DepartmentDTO> getAll() {
+        List<DepartmentDTO> departmentList = new ArrayList<>();
+        departmentRepository.findAll().forEach(department -> departmentList.add(convertENTtoDTO(department)));
+
+        if (departmentList.isEmpty()) {
+            return Collections.emptyList();
+        }
+        return departmentList;
+    }
+
+    public DepartmentDTO getById(long id) {
+        Optional<Department> departmentById = departmentRepository.findById(id);
+        if (departmentById.isPresent()) {
+            return convertENTtoDTO(departmentById.get());
+        }
+        throw new DepartmentNotFoundException("Department with id " + id + "not found");
+    }
+
+    public DepartmentDTO add(final DepartmentDTO department) {
+        departmentRepository.save(convertDTOtoENT(department));
+        return department;
```

Comment:
No need to return entire object, just created ID

Reply:
![image](https://user-images.githubusercontent.com/76517975/196684228-b74b2703-9feb-4c3b-bd8d-ae01635d5e53.png)

Reply:
I must return the newly added employee, or I'm wrong ?

Reply:
Ok, usually the ID is just returned and by ID then you can verify if it exist, but if requirements are like this let it be

Reply:
But if you stick to that requirement I don't see anything returned from repo. You need to return updated object that contains Id, not just the same object you pass to repository

Reply:
Done